api/server/router/container/container_routes.go

```diff
@@ -532,12 +534,71 @@ func (s *containerRouter) postContainersAttach(ctx context.Context, w http.Respo
 		return errdefs.InvalidParameter(errors.Errorf("error attaching to container %s, hijack connection missing", containerName))
 	}
 
+	ctx, cancelStreams := context.WithCancel(ctx)
+	defer cancelStreams()
+
 	setupStreams := func() (io.ReadCloser, io.Writer, io.Writer, error) {
 		conn, _, err := hijacker.Hijack()
 		if err != nil {
 			return nil, nil, nil, err
 		}
 
+		func() {
```

Comment:
Hmm, the context from the request should be closed when the connection closes. Is this sufficient?

Reply:
Oh maybe not because of hijack...